ve-sched: nr-unint summing negative

From: Konstantin Khlebnikov <khlebnikov@openvz.org>

summing per-cpu counters can produce negative results, this break ve load-average calculation.

v2:
fixup for nr-running not needed

http://bugzilla.openvz.org/show_bug.cgi?id=1449

Signed-off-by: Konstantin Khlebnikov <khlebnikov@openvz.org>
---

 kernel/sched.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/kernel/sched.c b/kernel/sched.c
index 230c847..eadfd64 100644
--- a/kernel/sched.c
+++ b/kernel/sched.c
@@ -3282,6 +3282,8 @@ unsigned int nr_unint_vsched(struct vcpu_scheduler *vsched)
 
 	sum += (unsigned int)atomic_read(&vsched->nr_unint_fixup);
 #endif
+	if (unlikely((int)sum < 0))
+		sum = 0;
 	return sum;
 }
 
